### DISCUSSION TITLE
Self Hosted Forums / Messaging killer app?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3450] [Ideas]
My morning thoughts about a self-hosted messaging module. 
[https://www.marks.wiki/p/shift-a-killer-app-for-oqtane ](https://www.marks.wiki/p/shift-a-killer-app-for-oqtane )

Conceptualizing Oqtane SHIFT
During our discussions at Oqtane, when the idea of an enterprise messaging 'killer app' surfaced, I was struck by a moment of inspired curiosity: "How hard could it be?" This led to the playful yet potent branding of Oqtane SHIFT: Self Hosted Instant messaging For Teams—a name that encapsulates the initiative's playful spirit and serious ambition.

Implementing Oqtane SHIFT
The implementation of Oqtane SHIFT could be remarkably straightforward. Oqtane's architecture, with its intrinsic page-level security, provides a natural foundation for forums, which are essentially pages within the system. Integration with corporate SSO is seamless, and by adding a SHIFT module to a site page, enterprises could instantly enable an infinitely threaded forum—efficiency meets elegance.

Notification Systems within Oqtane SHIFT
Furthermore, Oqtane's built-in features pave the way for sophisticated communication flows. The platform already accommodates email notifications, so tagging someone in a SHIFT forum could trigger both an email and an in-app alert. To streamline this, we envision two separate yet complementary modules: one for the notification badge and another for in-app notifications—simple, yet effective additions that would enhance user interaction within the platform.

**Comment by sbwalker** [comment]
Since Oqtane is based on similar concepts to DNN, there is an opportunity to migrate an open source Forum project such as:

https://github.com/DNNCommunity/Dnn.CommunityForums

**Comment by rchibnik** [comment]
I am interested in hosting forums on Oqtane - I would be happy to contribute to a project. Not sure I'm up to building it myself

A port of Dnn.CommunityForums sounds like a promising aproach.
